apps/desktop/src/components/KeysForm.svelte

```diff
@@ -13,66 +14,60 @@
 	import Link from '@gitbutler/ui/link/Link.svelte';
 	import { onMount } from 'svelte';
 
-	const project = $state(getContext(Project));
-
-	const baseBranch = maybeGetContext(BaseBranch);
-	const projectsService = getContext(ProjectsService);
-
 	interface Props {
 		// Used by credential checker before target branch set
+		projectId: string;
 		remoteName?: string;
 		branchName?: string;
 		showProjectName?: boolean;
 		disabled?: boolean;
 	}
 
 	const {
+		projectId,
 		remoteName = '',
 		branchName = '',
 		showProjectName = false,
 		disabled = false
 	}: Props = $props();
 
-	let credentialCheck = $state<CredentialCheck>();
+	const baseBranch = maybeGetContext(BaseBranch);
+	const projectsService = getContext(ProjectsService);
+	const projectResult = $derived(projectsService.getProject(projectId));
+	const project = $derived(projectResult.current.data);
 
-	let selectedType: KeyType = $state(
-		typeof project.preferred_key === 'string' ? project.preferred_key : 'local'
-	);
+	let credentialCheck = $state<CredentialCheck>();
 
-	let privateKeyPath = $state(
-		typeof project.preferred_key === 'string' ? '' : project.preferred_key.local.private_key_path
+	let selectedType: KeyType = $derived(
+		typeof project?.preferred_key === 'string' ? project?.preferred_key : 'local'
 	);
 
-	function setLocalKey() {
-		if (privateKeyPath.trim().length === 0) return;
-		updateKey({
-			preferred_key: {
-				local: {
-					private_key_path: privateKeyPath.trim()
-				}
-			}
-		});
-	}
+	let privateKeyPath = $state('');
 
-	async function updateKey(detail: { preferred_key: Key }) {
+	async function updateKey(detail: { preferred_key: AuthKey }) {
 		try {
-			project.preferred_key = detail.preferred_key;
-			projectsService.updateProject(project);
+			if (project) {
+				projectsService.updateProject({ ...project, preferred_key: detail.preferred_key });
+			}
 		} catch (err: any) {
 			showError('Failed to update key', err);
 		}
 	}
 
 	let form = $state<HTMLFormElement>();
 
-	function onFormChange(form: HTMLFormElement) {
+	function onFormChange(selectedType: string) {
 		credentialCheck?.reset();
-		const formData = new FormData(form);
-		selectedType = formData.get('credentialType') as KeyType;
 		if (selectedType !== 'local') {
-			updateKey({ preferred_key: selectedType });
+			updateKey({ preferred_key: selectedType as AuthKey });
```

Comment:
Should this not be as KeyType?

Reply:
Does splitting the FormData handling into both the event handler and here actually make it clearer?